mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearRegressionWithSGD.scala

```diff
@@ -47,6 +52,7 @@ class StreamingLinearRegressionWithSGD private[mllib] (
     private var numIterations: Int,
     private var miniBatchFraction: Double)
   extends StreamingLinearAlgorithm[LinearRegressionModel, LinearRegressionWithSGD]
+  with StreamingDecaySetter[StreamingLinearRegressionWithSGD]
```

Comment:
Thanks for the suggestions. I think `A` captures the class of non-streaming algorithm. For example, 
``` scala
class StreamingLinearRegressionWithSGD 
  extends StreamingLinearAlgorithm[LinearRegressionModel, LinearRegressionWithSGD]
```
The concrete subtype that `A` capture is `LinearRegressionWithSGD`. However, the fluent setter needs to return `StreamingLinearRegressionWithSGD`, though.

What do you think?